src/routes/(app)/product/[id]/+page.server.ts

```diff
@@ -166,15 +166,30 @@ async function addToCart({ params, request, locals }: RequestEvent) {
 			chosenVariations: z.record(z.string(), z.string()).optional()
 		})
 		.parse(json);
+	const variationPrice =
+		product.hasVariations && chosenVariations
+			? Object.entries(chosenVariations).reduce((total, variation) => {
+					return (
+						total +
+						(product.variations?.find(
+							(vari) => vari.name === variation[0] && variation[1] === vari.value
+						)?.price || 0)
+					);
+			  }, 0)
+			: 0;
 
 	const customPrice =
 		customPriceAmount && customPriceCurrency
 			? {
 					amount: parsePriceAmount(customPriceAmount, customPriceCurrency),
 					currency: customPriceCurrency
 			  }
+			: variationPrice > 0
+			? {
+					amount: parsePriceAmount(variationPrice.toString(), runtimeConfig.mainCurrency),
```

Comment:
`parsePriceAmount` is only to parse data passed from the frontend. It's not needed here